drivers/serial/uart_native_posix.c

```diff
@@ -261,9 +272,20 @@ static void np_uart_poll_out(const struct device *dev,
 				      unsigned char out_char)
 {
 	int ret;
-	struct native_uart_status *d;
+	struct native_uart_status *d = (struct native_uart_status *)dev->data;
+	struct pollfd pfd = { .fd = d->out_fd, .events = POLLHUP };
```

Comment:
I'd move this into the if block:
```diff
        int ret;
        struct native_uart_status *d = (struct native_uart_status *)dev->data;
-       struct pollfd pfd = { .fd = d->out_fd, .events = POLLHUP };
 
        if (wait_pts) {
+               struct pollfd pfd = { .fd = d->out_fd, .events = POLLHUP };
+
```
That makes it easier for the compiler to optimize it out (and for humans to know where it is used)